### PR TITLE
Update endpoint registry

### PR DESCRIPTION
**Pull request by fazlan-nazeem**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/impl/RegistriesApiServiceImpl.java [hunk 1]
```diff
@@ -164,7 +164,23 @@ public Response createRegistryEntry(String registryId, RegistryEntryDTO registry
...
+            if (endpointRegistry == null) {
+                RestApiUtil.handleResourceNotFoundError("Endpoint registry with the id: " + registryId +
+                        " is not found", log);
+            }
+            registryProvider.updateEndpointRegistry(registryId, registryToUpdate);
+        } catch (APIManagementException e) {
+            RestApiUtil.handleInternalServerError("Error while updating the endpoint registry " +
+                    "with id: " + registryId, e, log);
+        }
+        return Response.ok().entity("Successfully updated the endpoint registry").build();
```
Comment (vithu30) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/impl/RegistriesApiServiceImpl.java:183]: Shall we return the updated Registry object in response.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 2]
```diff
@@ -14757,6 +14757,37 @@ public String addEndpointRegistry(EndpointRegistryInfo endpointRegistry, int ten
...
+     * @param registryId uuid of the endpoint registry
+     * @param endpointRegistry EndpointRegistryInfo object with updated details
+     * @return uuid of the updated endpoint registry
+     * @throws APIManagementException if unable to update the endpoint registry
+     */
+    public String updateEndpointRegistry(String registryId, EndpointRegistryInfo endpointRegistry) throws
+            APIManagementException {
+
+        String query = SQLConstants.UPDATE_ENDPOINT_REGISTRY_SQL;
+        String uuid = UUID.randomUUID().toString();
```
Comment (vithu30) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:14772]: We don't need to generate uuid when updating the Registry. Shall we remove it?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/impl/RegistriesApiServiceImpl.java (hunk 1 above)
Comment (fazlan-nazeem) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/impl/RegistriesApiServiceImpl.java:183]: ack. missed it

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java (hunk 2 above)
Comment (fazlan-nazeem) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:14772]: ack

**Pull request merged**